crates/volta-migrate/src/v4.rs

```diff
@@ -0,0 +1,165 @@
+use std::convert::TryFrom;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+use crate::empty::Empty;
+use crate::v3::V3;
+use log::{debug, warn};
+use volta_core::error::{Context, ErrorKind, Fallible, VoltaError};
+use volta_core::fs::remove_file_if_exists;
+use volta_core::platform::PlatformSpec;
+use volta_core::session::Session;
+use volta_core::tool::{Package, PackageConfig};
+use volta_core::version::VersionSpec;
+use volta_layout::{v3, v4};
+use walkdir::WalkDir;
+
+mod config;
+
+use config::LegacyPackageConfig;
+
+/// Represents a V4 Volta layout (used by Volta v0.9.0 and above)
```

Comment:
nit: We should update this comment to reflect the expected version (1.1.2) rather than 0.9.0